### PR TITLE
split rootless local and remote testing

### PR DESCRIPTION
**Pull request by baude**
Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

**Comment by baude** [comment]
It looks like this really helped standardize test times.  Remember this is without the caching benefits as it hadn't been merged yet and I have not rebased due to test issues:  Here are some results:
```
rootless
  local 18:47
  remote 20:99
in_podmn
  local 24:17
f28
  local 20:57
  remote 20:53
f29
  local 21:44
  remote 21:22
ubuntu
  local 21:48
  remote 25:08
```
And I believe that caching cut the local test times in just about one-half.

**Comment by mheon** [comment]
LGTM

File: contrib/cirrus/setup_container_environment.sh [hunk 1]
```diff
@@ -5,15 +5,12 @@ source $(dirname $0)/lib.sh
...
 # test cases are actually run. As CRIU uses iptables to lock and unlock
 # the network during checkpoint and restore it needs the following two
 # modules loaded.
 modprobe ip6table_nat || :
 modprobe iptable_nat || :
 
-# Build the test image
-${CONTAINER_RUNTIME} build -t ${IMAGE} -f Dockerfile.${DIST} .
+# Pull the test image
+${CONTAINER_RUNTIME} pull quay.io/libpod/in_podman:latest
```
Comment (cevich) [inline contrib/cirrus/setup_container_environment.sh]: Is it possible to define `quay.io/libpod/in_podman:latest` in `lib.sh` then reference the variable instead?  That would be a lot cleaner and easier to maintain.

Comment (baude) [reply contrib/cirrus/setup_container_environment.sh]: if you really want to twist my arm on that, yes

Comment (cevich) [reply contrib/cirrus/setup_container_environment.sh]: /me gets out his plush wrench...puts it back in it's hiding spot, and fetches the iron one :grin:

**Comment by cevich** [comment]
Oh, crap...I [almost forgot: *the docs* :confused: ](https://github.com/containers/libpod/blob/master/contrib/cirrus/README.md#special_testing-task)

**Pull request merged**